testcases/InstallHostOS.py

```diff
@@ -0,0 +1,152 @@
+            ks_url = 'http://%s:%s/%s' % (my_ip, port, ks)
+            kernel_args = "ifname=net0:%s ip=%s::%s:%s:%s:net0:none nameserver=%s inst.ks=%s" % (host_mac_addr,
+                                                                                                 self.host.ip,
+                                                                                                 self.conf.args.host_gateway,
+                                                                                                 self.conf.args.host_submask,
+                                                                                                 self.host.hostname(),
+                                                                                                 self.conf.args.host_dns,
+                                                                                                 ks_url)
+            self.system.goto_state(OpSystemState.PETITBOOT_SHELL)
+            cmd = "[ -f %s ]&& rm -f %s;[ -f %s ] && rm -f %s;true" % (vmlinux,
+                                                                       vmlinux,
+                                                                       initrd,
+                                                                       initrd)
+            self.c.run_command(cmd)
+            self.c.run_command("wget http://%s:%s/%s" % (my_ip, port, vmlinux))
+            self.c.run_command("wget http://%s:%s/%s" % (my_ip, port, initrd))
+            self.c.run_command("kexec -i %s -c \"%s\" %s -l" % (initrd,
+                                                                kernel_args,
+                                                                vmlinux))
+            self.c.get_console().send("kexec -e\n")
+        else:
+            pass
+        # Do things
+        rawc = self.c.get_console()
+        rawc.expect('opal: OPAL detected', timeout=60)
+        r = None
+        while r != 0:
+            r = rawc.expect(['Running post-installation scripts',
+                             'Starting installer',
+                             'Setting up the installation environment',
+                             'Starting package installation process',
+                             'Performing post-installation setup tasks',
+                             'Configuring installed system'], timeout=600)
+        rawc.expect('reboot: Restarting system', timeout=300)
+        self.system.set_state(OpSystemState.IPLing)
+        self.system.goto_state(OpSystemState.PETITBOOT_SHELL)
+        OpTestInstallUtil.stop_server()
+        OpTestInstallUtil.set_bootable_disk(self.host.get_scratch_disk())
+        self.system.goto_state(OpSystemState.OS)
+        con = self.system.sys_get_ipmi_console()
+        self.system.host_console_login()
+        self.system.host_console_unique_prompt()
+        con.run_command("uname -a")
+        con.run_command("cat /etc/os-release")
+        # Run additional host commands if any from user
```

Comment:
we can remove this running commands from the user, those above two commands(uname and osrelease) are sufficient. You can add the `host_gather_opal_msg_log` and `host_gather_kernel_log` functions here to capture the debug logs after installation.

Reply:
cool, I will add those functions, the user command is needed as we might need to run some commands to upgrade host, with additional repo, those would be adhoc, this will help us to hook it later...